src/iidm/NetworkVariant.cpp

```diff
@@ -16,17 +18,37 @@ namespace iidm {
 namespace network {
 
 VariantImpl::VariantImpl(Network& network) :
-    Variant(network) {
+    Variant(network),
+    m_connectedComponentsManager(network),
+    m_synchronousComponentsManager(network) {
 }
 
-VariantImpl::VariantImpl(Network& network, VariantImpl&& /*variant*/) noexcept :
-    Variant(network) {
+VariantImpl::VariantImpl(Network& network, VariantImpl&& variant) noexcept :
+    Variant(network),
+    m_connectedComponentsManager(network, std::move(variant.m_connectedComponentsManager)),
+    m_synchronousComponentsManager(network, std::move(variant.m_synchronousComponentsManager)) {
 }
 
 std::unique_ptr<VariantImpl> VariantImpl::copy() const {
     return stdcxx::make_unique<VariantImpl>(m_owner);
```

Comment:
To be discussed : I know that components managers will recompute components if needed later, but the fact that `copy` does not copy components managers may be a bit disturbing.

Reply:
Same in java, so I propose to keep like this even if I agree with you

Reply:
https://github.com/powsybl/powsybl-core/issues/1548